tests/integration2/pilot/sidecar_api_test.go

```diff
@@ -0,0 +1,134 @@
+package pilot
+
+import (
+	"fmt"
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/api/components"
+	"istio.io/istio/pkg/test/framework/api/descriptors"
+	"istio.io/istio/pkg/test/framework/api/ids"
+	"istio.io/istio/pkg/test/framework/api/lifecycle"
+	"istio.io/istio/pkg/test/framework/runtime/components/environment/native"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/structpath"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestSidecarListeners(t *testing.T) {
+	// Call Requires to explicitly initialize dependencies that the test needs.
+	ctx := framework.GetContext(t)
+	scopes.Framework.SetOutputLevel(log.DebugLevel)
+
+	// TODO: Limit to Native environment until the Kubernetes environment is supported in the Galley
+	// component
+	ctx.RequireOrSkip(t, lifecycle.Test, &descriptors.NativeEnvironment)
+	ctx.RequireOrFail(t, lifecycle.Test, &ids.Galley)
+	ctx.RequireOrFail(t, lifecycle.Test, &ids.Pilot)
+
+	// Load the canonical dataset into Galley and by implication Pilot
+	gal := components.GetGalley(ctx, t)
+	env, err := native.GetEnvironment(ctx)
```

Comment:
What API changes do we need in order to make this platform agnostic? Maybe components can provide their expected Envoy listeners? Something more clever?

Reply:
I think components providing their endpoint & service metadata would likely be sufficient encapsulation. I can look into doing this.

Reply:
Done